File: Creational/Factory Method/lib/ConcreteFactoryB.cpp
#include "../inc/ConcreteFactoryB.hpp"
#include "../inc/ConcreteProductB.hpp"

ConcreteFactoryB::ConcreteFactoryB()
{

}

ConcreteFactoryB::~ConcreteFactoryB()
{

}
	
std::shared_ptr<AbstractProduct> ConcreteFactoryB::factory_method() const
{
	return std::make_shared<ConcreteProductB>();
};

File: Creational/Factory Method/Golang/ConcreteFactoryB.go
package main

type ConcreteFactoryB struct{

}

func (ConcreteFactoryB) factoryMethod() AbstractProduct{
	return &ConcreteProductB{}
}

func (concreteFactoryB ConcreteFactoryB) operation(){
	var abstractProduct AbstractProduct = concreteFactoryB.factoryMethod()
	abstractProduct.operation()
}